src/SelectField/index.js

```diff
@@ -1,145 +1,96 @@
-import PropTypes from 'prop-types'
+import propTypes from 'prop-types'
 import React, { Fragment } from 'react'
 import css from 'styled-jsx/css'
 import cx from 'classnames'
 
-import { ArrowUp, ArrowDown } from '../icons/Arrow'
-import { Valid, Warning, Error } from '../icons/Status'
-import { colors, fonts } from '../theme'
-import { createIcon } from '../icons/helpers'
-import {
-    iconStatusPropType,
-    iconStatuses,
-    statusToIcon,
-} from '../icons/constants'
+import { Arrow } from './Arrow'
+import { LabelFilled, LabelOutlined } from '../FieldLabel'
+import { Select } from './Select'
+import { colors } from '../theme'
+import { iconStatusPropType, iconStatuses } from '../icons/constants'
 import { inputKinds, inputSizes } from '../forms/constants'
-import { isPointInRect } from '../utils/math'
 import Help from '../Help'
-import Menu from '../Menu'
-import styles, { arrowIcon, menuOverride, selectIconStyles } from './styles'
 
-function createTrailIcon(status, trail, fn) {
-    const icon = status !== iconStatuses.DEFAULT ? statusToIcon[status] : trail
-    const options = {
-        action: fn,
-        className: selectIconStyles.className,
+const styles = css`
+    .base {
+        display: inline-block;
+        position: relative;
+        width: 100%;
+        background-color: inherit;
+        color: ${colors.grey700};
+        pointer-events: all;
+        user-select: none;
     }
 
-    return createIcon(icon, options)
-}
-
-function markActive(list, value) {
-    if (!value) {
-        return list
+    .disabled {
+        cursor: not-allowed;
+        opacity: 1;
     }
-
-    return list.slice(0).map(item => {
-        item.active = item.value && item.value === value
-
-        if (Array.isArray(item.list)) {
-            item.list = markActive(item.list, value)
-        }
-
-        return item
-    })
-}
+`
 
 class SelectField extends React.Component {
-    state = {
-        open: false,
-        labelWidth: 0,
-    }
+    elContainer = React.createRef()
 
     constructor(props) {
         super(props)
-        this.labelRef = React.createRef()
+
+        this.state = {
+            focused: props.focused,
+            open: false,
+        }
     }
 
     componentDidMount() {
         document.addEventListener('click', this.onDocClick)
-
-        this.setState({ labelWidth: this.labelRef.current.offsetWidth })
     }
 
     componentWillUnmount() {
         document.removeEventListener('click', this.onDocClick)
     }
 
     onDocClick = evt => {
```

Comment:
In the new implementation, why do we need to keep track of the document clicks? Just tapping into the focus and blur event on the select element should be enough....

Reply:
Probably should have been to keep track of if the label on select fields should shrink or not, but that feature is broken so I'll clean up the code and remove it.

Reply:
78439ce